### PR TITLE
add rest table

### PR DESCRIPTION
**Pull request by gkorland**
## Summary by CodeRabbit

- **New Features**
	- Comprehensive documentation for the REST API has been introduced, detailing endpoints for user authentication, configuration management, user management, and graph management.
	- New endpoints added for user authentication (login/logout), user management (create/delete/retrieve users), and graph management (create/delete/query graphs and schemas).

- **Documentation**
	- Enhanced API documentation in a structured format for easier understanding and usage by developers.
	- Expanded wordlist with relevant programming terms and functionalities to support development efforts.

### CODE REVIEW

File: integration/rest.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+title: "Rest API"
+description: >
+ Rest API detailed doc
+parent: "integration"
+---
+
+| Request Name  | Request Endpoint | Request Method | Request Headers | Request Parameters | Body | Successful Response | Description|
+| ------------- | ---------------- | -------------- | --------------- | ------------------ | ---- | ------------------- | ---------- |
+| Login | [/api/auth/providers](http://localhost:3000/api/auth/providers)  | GET | None | None | None | Status Code: 200<br>{<br>"credentials": {<br>"id": "credentials",<br>"name": "Credentials",<br>"type": "credentials",<br>"signinUrl": "http://localhost:3000/api/auth/signin/credentials",<br>"callbackUrl": "http://localhost:3000/api/auth/callback/credentials"<br>}<br>} | This request is used for user authentication. |
```
[the remark this answers is not in the document]
Comment (gkorland) [reply integration/rest.md:10]: not relevant

File: integration/rest.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+| ------------- | ---------------- | -------------- | --------------- | ------------------ | ---- | ------------------- | ---------- |
+| Login         | [/api/auth/providers](http://localhost:3000/api/auth/providers)                                                                             | GET            | None                                            | None                                                                                                  | None                                                                                                                                       | Status Code: 200<br>{<br>"credentials": {<br>"id": "credentials",<br>"name": "Credentials",<br>"type": "credentials",<br>"signinUrl": "http://localhost:3000/api/auth/signin/credentials",<br>"callbackUrl": "http://localhost:3000/api/auth/callback/credentials"<br>}<br>} | This request is used for user authentication.                                                                                                                                                                                                                                                                               |
```
[the remark this answers is not in the document]
Comment (gkorland) [reply integration/rest.md]: not relevant

File: integration/rest.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+| Delete A Schema       | [http://localhost:3000/api/graph/schemaName](http://localhost:3000/api/graph/schemaName)                                                                         | DELETE         | Authorization: Cookie                           | 1\. schemaName                                                                                        | None                                                                                                                                       | Status Code: 200<br>{<br>"message": "schemaName graph deleted"<br>}                                                                                                                                                                                                          | This request deletes the schema with the specified name. Authorization is required via a cookie in the headers. The response confirms the deletion of the schema.                                                                                                                                                           |
```
[the remark this answers is not in the document]
Comment (gkorland) [reply integration/rest.md]: not relevant

File: integration/rest.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (gkorland) [reply integration/rest.md:10]: not relevant

File: integration/rest.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (gkorland) [reply integration/rest.md]: not relevant

**Review by barakb — APPROVED** [review]

**Review by barakb — APPROVED** [review]

**Review by barakb — APPROVED** [review]

**Pull request merged**